lms/services/vitalsource/client.py

```diff
@@ -13,11 +16,43 @@ def __init__(self, lti_launch_key, lti_launch_secret):
         :type lti_launch_secret: str
         :raises ValueError: If credentials are invalid
         """
-        if not lti_launch_key or not lti_launch_secret:
-            raise ValueError("VitalSource LTI launch credentials are invalid")
+        if not all([lti_launch_key, lti_launch_secret, api_key]):
+            raise ValueError("VitalSource credentials are invalid")
+
+        self._http_service = http_service
 
         self._lti_launch_key = lti_launch_key
         self._lti_launch_secret = lti_launch_secret
+        self._api_key = api_key
+
+    def _api_get(self, endpoint):
+        url = f"https://api.vitalsource.com/v4/{endpoint}"
+
+        return self._http_service.get(
+            url, headers={"X-VitalSource-API-Key": self._api_key}
+        )
+
+    def book_info(self, book_id: str):
+        try:
+            response = self._api_get(f"products/{book_id}")
```

Comment:
Are you validating the format of a book ID anywhere? If not the `book_id` could potentially include characters that require encoding. We do validate it against a regex on the frontend (which IIRC allows only upper-case ASCII letters and dashes) but you shouldn't trust that on the backend.

Reply:
Now the book_id format is checked against a regexp on the view @robertknight